src/webgpu/web_platform/external_texture/video.spec.ts

```diff
@@ -3,13 +3,150 @@ Tests for external textures from HTMLVideoElement (and other video-type sources?
 
 - videos with various encodings, color spaces, metadata
 
-TODO: get test videos from WebGL CTS
-TODO(kainino0x): (and set up the build to deal with data files)
 TODO: consider whether external_texture and copyToTexture video tests should be in the same file
-TODO: plan
+
+Test Coverage:
+  - Tests that we can import an HTMLVideoElement into a GPUExternalTexture and sample from it for:
+      - All supported video formats {vp8, vp9, ogg, mp4}
+      TODO: Multiplanar scenarios
 `;
 
 import { makeTestGroup } from '../../../common/framework/test_group.js';
 import { GPUTest } from '../../gpu_test.js';
 
+const kHeight = 16;
+const kWidth = 16;
+const kFormat = 'rgba8unorm';
+const kVideoSources: string[] = [
+  'red-green.webmvp8.webm',
+  'red-green.bt601.vp9.webm',
+  'red-green.mp4',
+  'red-green.theora.ogv',
+];
+
 export const g = makeTestGroup(GPUTest);
+
+g.test('importExternalTexture,sample')
+  .desc(
+    `
+Test that importing different video formats into an external texture and sampling from them works correctly.
+`
+  )
+  .params(u =>
+    u //
+      .combine('videoSource', kVideoSources)
+  )
+  .fn(async t => {
+    const videoUrl = '../out/resources/' + t.params.videoSource;
+    const video = document.createElement('video');
+    video.src = videoUrl;
+    if (!video.canPlayType) {
+      throw new Error();
+    }
+
+    video.loop = true;
+    video.muted = true;
+    video.play();
+    await video.play();
```

Comment:
Is it necessary to call this twice? One `await video.play()` seems enough, but if it's not, leave a comment about it.

Reply:
This was a mistake. Only one is needed.